### PR TITLE
[examples] Add examples for STM32F103C8T6 Minimum System Development …

### PR DESCRIPTION
**Pull request by strongly-typed**
…Board.

### CODE REVIEW

**Comment by salkinium** [comment]
Is this the "blue pill" board from eBay?

**Comment by salkinium** [comment]
Please fix the commit message length ;-P

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include <xpcc/architecture/platform.hpp>
+
+using namespace xpcc::stm32;
+
+
+namespace Board
+{
+
+/// STM32F103 running at 72MHz generated from the external 8MHz crystal
+using systemClock = SystemClock<Pll<ExternalCrystal<MHz8>, MHz72> >;
```
Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp]: This works only by accident! `SystemClock` is only configured for the F100 chip.

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+using LedGreen = GpioOutputC13;		// User LED
+
```
Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp]: Missing `Leds` SoftwareGpio and Button to be compatible with [Hello World](https://github.com/roboterclubaachen/getting-started-with-xpcc/blob/master/hello-world/main.cpp). (I know, needs to be documented by me!)

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/board.cfg [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[build]
+device = stm32f103c8t6
+
+[parameters]
+core.cortex.0.enable_hardfault_handler_led = true
+core.cortex.0.hardfault_handler_led_port = C
+core.cortex.0.hardfault_handler_led_pin = 13
+
+[openocd]
+configfile = board/stm32f103_msdb.cfg
```
Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/board.cfg]: Hm, good point. I need to implement local referencing of `.cfg` files inside the board folder by adding it to openocds search path.

**Comment by salkinium** [comment]
I think "blue_pill" is a better name for it.

**Comment by salkinium** [comment]
Can you split this commit into two? [One for adding the board](https://github.com/roboterclubaachen/xpcc/commit/8041aabaef47a86a9708ed9d7503aea21967c8bf) and [one for adding the example](https://github.com/roboterclubaachen/xpcc/commit/6c1a1117b7d97d164db4ca6f0ecc783ea161a962). This will make it easier to cherry pick the boards in future.

Please update the ReadMe with a link of the blue pill and the number of stars (4 stars) in the first commit.

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp (hunk 2 above)
Comment (strongly-typed) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp]: The only button on the Blue Pill is the reset button. So I use GpioUnused.

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp (hunk 1 above)
Comment (strongly-typed) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp]: So what to do?

**Comment by strongly-typed** [comment]
Yes, it's the "Blue Pill". 
[VCC-GND.com](vcc-gnd.com) is dead, so linked to [STM32Duino](http://wiki.stm32duino.com/index.php?title=Blue_Pill)

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp (hunk 1 above)
Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp]: You may be able to [use the F103RB clock tree from the Nucleo board](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/board/nucleo_f103rb/nucleo_f103rb.hpp#L27-L87) and modify it to the C8 clock tree.

File: src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp (hunk 2 above)
Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f103c8t6_msdb/stm32f103c8t6_msdb.hpp]: 👍 This is also done on [the Arduino board](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/board/arduino_uno/arduino_uno.hpp#L50).

**Comment by strongly-typed** [comment]
@salkinium Are you fine with all these commits or did I miss something?

**Comment by salkinium** [comment]
The system clock should be implemented manually. [Reasoning](https://github.com/roboterclubaachen/xpcc/issues/139). I know it's ugly, but this way it's a lot less magic for devs who want to port to another board.

The way I figure out the peripherals is to look at the reference manual section 3.3 (memory map), which gives you the peripheral and the attached bus. You can use the F103RB board target as a starting point and modify it.

**Comment by strongly-typed** [comment]
Oh, you implicitly and politely asked me to do that?
Sorry, missed it and I look into it now.

**Comment by salkinium** [comment]
> Oh, you implicitly and politely asked me to do that?

I think the UK is getting to me…

**Comment by salkinium** [comment]
Awesome, will merge manually.

**Pull request closed without merging**